subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/resolver/MavenMetadataLoader.java

```diff
@@ -53,8 +59,14 @@ public MavenMetadata load(URI metadataLocation) throws ResourceException {
         return metadata;
     }
 
-    private void parseMavenMetadataInfo(final URI metadataLocation, final MavenMetadata metadata) {
-        ExternalResource resource = repository.getResource(metadataLocation, true);
+    private void parseMavenMetadataInfo(final URI metadataLocation, final MavenMetadata metadata) throws IOException {
+        ExternalResource resource = cacheAwareExternalResourceAccessor.getResource(metadataLocation, new CacheAwareExternalResourceAccessor.ResourceFileStore() {
```

Comment:
why do we want to cache them? It's more likely faster to get the thing than check whether it's changed.